src/cdomains/intDomain.ml

```diff
@@ -101,13 +101,66 @@ module Size = struct (* size in bits as int, range as int64 *)
     try int64_of_big_int y with _ -> raise Not_in_int64
 end
 
-module Interval32 : S with type t = (int64 * int64) option = (* signed 32bit ints *)
+module IntPair =
+struct
+  type t = (int64 * int64) [@@deriving to_yojson]
+
+  let min_int, max_int = Size.range Cil.IInt
+
+  let compare (x0, y0) (x1, y1) =
+    match Int64.compare x0 x1 with
+    | 0 -> Int64.compare y0 y1
+    | c -> c
+
+  let leq (x: t option) (y:t option) =
+    match x, y with
+    | None, _ -> true
+    | Some _, None -> false
+    | Some (x1,x2), Some (y1,y2) -> Int64.compare x1 y1 >= 0 && Int64.compare x2 y2 <= 0
+
+  let norm = function None -> None | Some (x,y) ->
+    if Int64.compare x y > 0 then None
+    else if Int64.compare min_int x > 0 || Int64.compare max_int y < 0 then Some (min_int, max_int)
+    else Some (x,y)
+
+  let norm_set x =
+    match norm x with
+    | Some y -> y
+    | None -> (1L, 0L)
+
+  let add (x1, x2) (y1, y2) = ((Int64.add x1 y1), (Int64.add x2 y2))
+
+  let neg (x1, x2) _ = (Int64.neg x2, Int64.neg x1)
```

Comment:
What is the `_` good for here?

Reply:
The thing is, my `arithm` function (one that does all the pairwise computations) takes a binary function as an argument -- in order not to create another `arithm`, I just made the `neg` binary :D

Reply:
Hmm, that's a bit odd. Maybe rather do this where you pass something to this `arithm` like this `fun a _ -> neg a`. I think this is less confusing than adding another argument to neg